src/app/user/api/patchUser.js

```diff
@@ -39,19 +49,81 @@ const patchUser = async (req, res) => {
     return res.status(400).send(validationErrorMessage);
   }
 
+  const nameDetailsChanged = ['given_name', 'family_name'].some((property) => property in req.body && req.body[property].trim().toLowerCase() !== user[property].trim().toLowerCase());
+  const emailAddressChanged = !!req.body.email && req.body.email.trim().toLowerCase() !== user.email.trim().toLowerCase();
+
   // Patch user
   const updatedUser = Object.assign(userModel, req.body);
-  await update(updatedUser.sub, updatedUser.given_name, updatedUser.family_name,
-    updatedUser.email, updatedUser.job_title, updatedUser.phone_number, updatedUser.legacyUsernames, correlationId);
+  await update(
+    updatedUser.sub,
+    updatedUser.given_name,
+    updatedUser.family_name,
+    updatedUser.email,
+    updatedUser.job_title,
+    updatedUser.phone_number,
+    updatedUser.legacyUsernames,
+    correlationId,
+  );
 
-  if (config.toggles && config.toggles.notificationsEnabled) {
-    const serviceNotificationsClient = new ServiceNotificationsClient({
-      connectionString: config.notifications.connectionString,
-    });
-    const jobId = await serviceNotificationsClient.notifyUserUpdated(safeUser(updatedUser));
-    logger.info(`Send user updated notification for ${user.sub} with job id ${jobId} (reason: patch)`, { correlationId });
+  if (!!user.is_entra && !!user.entra_oid) {
+    let nameChangeFailed = false;
+    let emailChangeFailed = false;
+    const errorMessage = { type: 'error', message: 'Unable to update user record, there was an error with Entra' };
+
+    if (nameDetailsChanged === true) {
```

Comment:
My eyes keep mixing up the naming `nameChangeFailed` and `nameDetailsChanged`; what you have makes complete sense but for some reason I keep getting tripped on this naming.

Reply:
I can change the variable name(s) if you'd like?